integration-tests/src/main/java/org/apache/druid/testing/utils/KinesisAdminClient.java

```diff
@@ -156,15 +160,35 @@ public boolean verfiyPartitionCountUpdated(String streamName, int oldShardCount,
     return actualShardCount == oldShardCount + newShardCount;
   }
 
+  private List<Shard> listShards(String streamName)
+  {
+    ListShardsRequest listShardsRequest = new ListShardsRequest().withStreamName(streamName);
+    List<Shard> shards = new ArrayList<>();
+    while (true) {
+      ListShardsResult listShardsResult = amazonKinesis.listShards(listShardsRequest);
+      shards.addAll(listShardsResult.getShards());
+      String nextToken = listShardsResult.getNextToken();
+      if (nextToken == null) {
+        return shards;
+      }
+      listShardsRequest = new ListShardsRequest().withNextToken(listShardsResult.getNextToken());
+    }
+  }
 
-  private boolean verifyStreamStatus(StreamDescription streamDescription, StreamStatus streamStatusToCheck)
+  private boolean verifyStreamStatus(String streamName, StreamStatus... streamStatuses)
   {
-    return streamStatusToCheck.toString().equals(streamDescription.getStreamStatus());
+    String status = getStreamStatus(streamName);
+    for (StreamStatus streamStatus : streamStatuses) {
```

Comment:
nit: this loop can be `return Arrays.stream(streamStatuses).map(StreamStatus::toString).anyMatch(streamName::equals);`